src/app/user-page/user-page.component.ts

```diff
@@ -76,5 +84,14 @@ export class UserPageComponent extends Base implements OnInit {
 
   ngOnInit(): void {
     this.getUserInfo(this.username);
+    this.tokenQuery.tokens$.pipe(takeUntil(this.ngUnsubscribe)).subscribe((tokens: TokenUser[]) => {
+      for (const account of this.accountsInfo) {
+        const found = tokens.find((token) => token.tokenSource === account.source);
+        if (found && !this.publicAccountsSource.includes(account.source)) {
+          account.username = found?.username;
+          this.otherLinkedAccountsInfo.push(account);
```

Comment:
Should be something like `this.otherLinkedAccountsInfo.push(Object.assign({username: found?.username}), account)`, which will create a copy of the account object.